src/btree/bt_read.c

```diff
@@ -390,8 +388,8 @@ __page_read_lookaside(WT_SESSION_IMPL *session, WT_REF *ref,
 			WT_STAT_CONN_INCR(
 			    session, cache_read_lookaside_skipped);
 			ref->page_las->eviction_to_lookaside = true;
-			*final_statep = WT_REF_LIMBO;
 		}
+		*final_statep = WT_REF_LIMBO;
```

Comment:
@agorrod this seems like a straight bug: if we got here with a limbo page and decided not to read the history, it would transition to `WT_REF_MEM`.  Maybe this is unlikely because of the higher level checks on the state in `__wt_page_in_func` but I don't think it's impossible to be handling a page that is already `WT_REF_LIMBO` here, given the right interleaving.

Reply:
I agree - this looks like a bug fix. Nice catch.